Error deleting VirtualServer when hostGroup is used (no Virtual Server Address Found)

#### Setup Details
 CIS Version : 2.7.1
 Build: f5networks/k8s-bigip-ctlr:latest       
 BIGIP Version: Big IP 15.1
 AS3 Version: 3.33  
 Agent Mode: AS3/CCCL       
 Orchestration: K8S/OSCP      
 Orchestration Version:       
 Pool Mode: Cluster/Nodeport      
 Additional Setup details: <Platform/CNI Plugins/ cluster nodes/ etc>

#### Description  
When creating multiple VirtualServers with the same hostGroup and then try to delete them, on the last virtual server you will receive the following error on CIS
2022/02/04 09:21:17 [ERROR] Error in virtualserver address: no Virtual Server Address Found
On Kubernetes the VS CR is deleted but on the F5 the virtual server is still configured. 
Also the error keep on appearing on the logs 

2022/02/04 09:15:50 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:15:50 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:15:50 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:15:50 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:15:50 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:15:50 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:15:50 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:15:50 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:15:51 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:15:52 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:15:55 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:16:00 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:16:10 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:16:31 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:17:12 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:18:34 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:21:17 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:26:45 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:37:40 [ERROR] Error in virtualserver address: no Virtual Server Address Found
2022/02/04 09:54:20 [ERROR] Error in virtualserver address: no Virtual Server Address Found   

The only way to remove this error is to enabled ""--ipam=true" on the CIS arguments even if you are not using ipamLabel or IPAM in general

#### Steps To Reproduce
1) deploy the manifests I have uploaded (vs1, vs2 and CIS) 
2)deploy vs1 and vs2
3)delete vs1
4) delete vs2

uncomment the --ipam=true argument (redeploy CIS) and repeat the process. This time it should work
[cis-ctlr-cluster-crd.yml.txt](https://github.com/F5Networks/k8s-bigip-ctlr/files/8001949/cis-ctlr-cluster-crd.yml.txt)
[vs-2.yml.txt](https://github.com/F5Networks/k8s-bigip-ctlr/files/8001950/vs-2.yml.txt)
[vs-1.yml.txt](https://github.com/F5Networks/k8s-bigip-ctlr/files/8001951/vs-1.yml.txt)

--- a/docs/RELEASE-NOTES.rst
+++ b/docs/RELEASE-NOTES.rst
@@ -15,6 +15,7 @@ Enhancements
 * Added fix for processing oldest route when same host and path in routes
 * Added fix for cis crash with routes
 * :issues:`2212` Fix ExternalDNS adds both VSs to a Wide IP pool with using "httpTraffic: allow" with VS CR
+* :issues:`2222` Fix deleting VirtualServer using hostGroup
 
 2.7.1
 -------------

--- a/pkg/controller/worker.go
+++ b/pkg/controller/worker.go
@@ -803,6 +803,12 @@ func (ctlr *Controller) processVirtualServers(
 				log.Errorf("Error in virtualserver address: %s", err.Error())
 				return err
 			}
+			if ip == ""{
+				ip = virtual.Spec.VirtualServerAddress
+				if ip == ""{
+					return fmt.Errorf("No VirtualServer address found for: %s", virtual.Name)
+				}
+			}
 		}
 	}
 	// Depending on the ports defined, TLS type or Unsecured we will populate the resource config.
@@ -1138,7 +1144,7 @@ func getVirtualServerAddress(virtuals []*cisapiv1.VirtualServer) (string, error)
 			}
 		}
 	}
-	if vsa == "" {
+	if len(virtuals) != 0 && vsa == "" {
 		return "", fmt.Errorf("no Virtual Server Address Found")
 	}
 	return vsa, nil

